examples/jupyter/tensor_addition.ipynb

```diff
@@ -0,0 +1,226 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "id": "866d36f8",
+   "metadata": {},
+   "source": [
+    "### Installation"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "fa32231a",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# clone the wave github repo\n",
+    "!git clone https://github.com/iree-org/wave.git\n",
+    "# install ROCm Pytorch dependencies\n",
+    "!pip install -r pytorch-rocm-requirements.txt\n",
+    "# install wave and its dependencies\n",
+    "!pip install wave-lang"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "52993703",
+   "metadata": {},
+   "source": [
+    "### Wave Tensor/Matrix Addition Kernel Tutorial"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "e55e18af",
+   "metadata": {},
+   "source": [
+    "#### (1) Importing Libraries"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "b636039f",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import torch\n",
+    "\n",
+    "import wave_lang.kernel.wave as tkw\n",
+    "from wave_lang.kernel._support.dtype import f16\n",
+    "from wave_lang.kernel._support.indexing import sym\n",
+    "from wave_lang.kernel.lang.global_symbols import *\n",
+    "from wave_lang.kernel.lang.wave_types import *\n",
+    "from wave_lang.kernel.wave.compile import WaveCompileOptions, wave_compile\n",
+    "from wave_lang.kernel.wave.utils.run_utils import set_default_run_config"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "e169cc14",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Define symbolic dimensions for our matrices\n",
+    "M = sym.M # Rows of A, B and C\n",
+    "N = sym.N # Cols of A, B and C\n",
+    "\n",
+    "# Define workgroup tile sizes\n",
+    "BLOCK_M = sym.BLOCK_M\n",
+    "BLOCK_N = sym.BLOCK_N\n",
+    "\n",
+    "# Define the address space for our memory buffers\n",
+    "ADDRESS_SPACE_A = sym.ADDRESS_SPACE_A\n",
+    "ADDRESS_SPACE_B = sym.ADDRESS_SPACE_B\n",
+    "ADDRESS_SPACE_C = sym.ADDRESS_SPACE_C"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "d6f8cd53",
+   "metadata": {},
+   "source": [
+    "#### (2) Constraints\n",
+    "To specify how we want to distribute the different dimensions of our problem."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "16ef23d5",
+   "metadata": {},
+   "outputs": [],
+   "source": [
```

Comment:
Could you add some more explanation of what is going on? Also would be great to tie into Will's debug logger to show the data held by each wave/workgroup? What do you think?